aiida/cmdline/commands/cmd_import.py

```diff
@@ -25,6 +26,52 @@
 COMMENT_MODE = ['newest', 'overwrite']
 
 
+class ImportPath(click.Path):
+    """AiiDA Verdi implementation of Click's Path-type.
+    Introduce tests to distinguish whether the path is to a local file
+    or a URL string to a file that needs to be downloaded.
+    """
+
+    def __init__(self, **kwargs):
+        super(ImportPath, self).__init__(**kwargs)
+
+    def __call__(self, value, param=None, ctx=None):
+        if value is not None:
+            try:
+                res = super(ImportPath, self).convert(value, param, ctx)
+            except click.exceptions.BadParameter:
+                res = self.convert(value, param, ctx)
+            return res
+        return None
+
+    def convert(self, value, param, ctx):
+        """Do checks for possible URL path
+        Parameters :param:`readable` and :param:`exists` are always set to `True`.
+        """
+        from six.moves import urllib
+        from django.core.validators import URLValidator
+        from django.core.exceptions import ValidationError as URLValidationError
+        from socket import timeout
+
+        url = value
+        timeout_seconds = 10
+
+        try:
+            URLValidator()(url)
+        except URLValidationError:
+            self.fail('%s "%s" is not a valid URL or local %s.' % (self.path_type, filename_to_ui(url), self.name),
```

Comment:
why "or local"?

Reply:
Because if it gets to this point, we simply assume that the input is a URL, but it may just be a mistyped local path. So to not completely confuse the user, i put in "or local".

Reply:
You're right. In this case, what about the following error message:
```
'"%s" is neither a valid filepath nor a valid URL.' % (filename_to_ui(url))
```

Reply:
if you want to make things more "analogous" between checking whether it is a path or a URL, you can do a 
```
try:
   ...
except: 
    pass
```
for both and at the very end of the function fail with this error message.

Reply:
That's a nice idea.